### PR TITLE
Add multi thread support for tests

### PR DESCRIPTION
**Pull request by birdstorm**
Add multi-thread for faster integration tests

### CODE REVIEW

**Comment by birdstorm** [comment]
refer to #158 

PTAL @Novemser

**Comment by Novemser** [comment]
@birdstorm Is this PR ready for review?

**Comment by birdstorm** [comment]
@Novemser yes, PTAL

File: integtest/src/main/scala/com/pingcap/spark/TestCase.scala [hunk 1]
```diff
@@ -57,11 +57,17 @@ class TestCase(val prop: Properties) extends LazyLogging {
 
   private val eps = 1.0e-2
 
-  protected var testsFailed = 0
-  protected var testsExecuted = 0
-  protected var testsSkipped = 0
-  protected var inlineSQLNumber = 0
-  protected var ignoredTest = 0
+  @volatile protected var testsFailed = 0
```
Comment (ilovesoup) [inline integtest/src/main/scala/com/pingcap/spark/TestCase.scala]: Atomic operation like a += 1 cannot be protected by volatile only.
a += 1 -> a = a + 1 involve both load and store 2 operations. When load, add and store, volatile does not ensure the original value you load is not changed by other threads.
Use AtomicInteger instead.

File: integtest/src/main/scala/com/pingcap/spark/DAGTestCase.scala [hunk 2]
```diff
@@ -106,21 +109,23 @@ class DAGTestCase(prop: Properties) extends TestCase(prop) {
...
+        testsExecuted += 1
+        inlineSQLNumber += 1
+        pool.submit(new SQLProcessorRunnable(sql, inlineSQLNumber, dbName))
+        logger.info("Running num: " + count + " sql")
       } catch {
         case _: Throwable => logger.error("result: Run SQL " + sql + " Failed!")
       }
     }
-    result = !result
+    pool.shutdown()
```
Comment (ilovesoup) [inline integtest/src/main/scala/com/pingcap/spark/DAGTestCase.scala]: How can you ensure tests done at this moment?

**Comment by Novemser** [comment]
Maybe we can first switch to scalatest, and then see if we need multi-threading to improve performance.

**Comment by birdstorm** [comment]
Closing this PR since we are switching to Scala Test Framework.

**Pull request closed without merging**